src/util/numbering.h

```diff
@@ -18,53 +18,42 @@ Author: Daniel Kroening, kroening@kroening.com
 #include <util/invariant.h>
 #include <util/optional.h>
 
-template <typename T>
-// NOLINTNEXTLINE(readability/identifiers)
-class numbering final
+/// \tparam Key a value type
+/// \tparam Map a metafunction which is invoked with two parameters to
+/// return an appropriate mapping class
+template <class Key, class Map>
+class template_numberingt final
 {
 public:
-  // NOLINTNEXTLINE(readability/identifiers)
-  typedef std::size_t number_type;
+  using number_type = std::size_t; // NOLINT
```

Comment:
It would be nice to not rewrite more than is strictly necessary.

Reply:
`using` is more flexible than `typedef` (I need to use the template form in a few places in this file), and the file is more internally consistent if I use `using` everywhere, so I think this is justified

Reply:
Ok, but do not that this is by far not the only change that seems to go beyond refactoring. I am not saying that any of those changes are wrong, but they are making the reviewer's life rather miserable.